com.unity.render-pipelines.universal/Editor/Camera/CameraStackReorderableList.cs

```diff
@@ -0,0 +1,301 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEditor.SceneManagement;
+using UnityEditorInternal;
+using UnityEngine;
+using UnityEngine.Rendering.Universal;
+
+namespace UnityEditor.Rendering.Universal
+{
+    internal class CameraStackReorderableList
+    {
+        static class Styles
+        {
+            public static GUIContent cameras = EditorGUIUtility.TrTextContent("Cameras", "The list of overlay cameras assigned to this camera.");
+        }
+
+        readonly ReorderableList m_LayerList;
+
+        UniversalRenderPipelineSerializedCamera serializedCamera { get; }
+        Camera camera { get; }
+        CameraEditor.Settings settings { get; }
+
+        #region Validation
+
+        public bool hasErrors => typeErrorCameras.Count > 0;
+        public string errorMessage =>
+            $"The type of the following Cameras must be Overlay render type: {string.Join(", ", typeErrorCameras.Select(c => c.name).Distinct())}.";
+        List<Camera> typeErrorCameras { get; } = new List<Camera>();
+
+        public bool hasWarnings => outputWarningCameras.Count > 0;
+        public string warningMessage =>
+            $"The output properties of this Camera do not match the output properties of the following Cameras: {string.Join(", ", outputWarningCameras.Select(c => c.name).Distinct())}.";
+        List<Camera> outputWarningCameras { get; } = new List<Camera>();
+
+        void ClearValidation()
+        {
+            outputWarningCameras.Clear();
+            typeErrorCameras.Clear();
+        }
+
+        static bool Validate(Camera cam, Func<bool> isValid, List<Camera> cameraList)
```

Comment:
I am confused by the naming here. so Validate will take a cam and compare it to a list with a validation functor.
my expectation was "if my cam isValid then add it to the list, if not remove it" however the code says: "if my condition is Valid then remove the cam from the list if possible and return true. however if my condition is not valid, add the camera to the list and return false"
I would have expected two functions (one for removal, one for adding)

Reply:
Modified with a better naming, thanks!